test/integration_test.rb

```diff
@@ -229,6 +235,46 @@ def test_mixed_borders
     end
     @workbook.apply_styles
     assert_equal 8, @workbook.styled_cells.count
-    assert_equal 6, @workbook.style_index.count
+    assert_equal 6, @workbook.styles.style_index.count
+  end
+
+  def test_merge_styles_1
+    @filename = 'merge_styles_1'
+    @workbook.add_worksheet do |sheet|
+      sheet.add_row
+      sheet.add_row ['', '1', '2', '3']
+      sheet.add_row ['', '4', '5', '6']
+      sheet.add_row ['', '7', '8', '9']
+      sheet.add_style 'B2:D4', b: true
+      sheet.add_style 'B2:D4', border: { style: :thin, color: '000000' }
+    end
+    @workbook.apply_styles
+    assert_equal 1, @workbook.styles.style_index.count
+  end
+
+  def test_merge_styles_2
+    @filename = 'merge_styles_2'
+    bold = @workbook.styles.add_style b: true
+
+    @workbook.add_worksheet do |sheet|
+      sheet.add_row ['A1', 'B1'], style: [nil, bold]
+      sheet.add_row ['A2', 'B2']
+      sheet.add_border 'A1:B2'
+    end
+    @workbook.apply_styles
```

Comment:
The border is added correctly here, but the bold face on B1 is getting lost.

![bold-issue](https://cloud.githubusercontent.com/assets/4750367/18030375/a4dec93c-6c81-11e6-86b3-71b93f4b92db.png)